cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -321,9 +321,17 @@ rebuildProjectConfig verbosity
                      }
                      cliConfig = do
 
+    progsearchpath <- liftIO $ getSystemSearchPath
+
+    let fileMonitorProjectConfig = newFileMonitor (distProjectCacheFile "config")
+
     fileMonitorProjectConfigKey <- do
       configPath <- getConfigFilePath projectConfigConfigFile
-      return (configPath, distProjectFile "")
+      return (configPath, distProjectFile "",
+                        (projectConfigHcFlavor, projectConfigHcPath, projectConfigHcPkg),
+                        progsearchpath,
```

Comment:
Does this mean that if I run `cabal` in a directory with a slightly modified `PATH`, then it's going to be a complete cache miss on everything? Even it doesn't change what `ghc` etc. gets picked? `direnv` users will be sad if so.

Reply:
My understanding is this will only be a cache miss on parsing and evaluating the conditionals in a project file, which should be pretty low-cost. But we certainly need to give this a workout to be sure.